### PR TITLE
Refactor ChargingEquipmentProfile to RechargingEquipmentProfile

### PR DESCRIPTION
**Pull request by skinkie**
@nick-knowles This is the style of pull request I would like to see from you. A **single change** that can be reviewed and evaluated independently.

To other reviewers: I have already made some typo fixes. I would again like to ask you if you can give this one a review, thanks for your precious time.

### CODE REVIEW

**Review by ue71603 — APPROVED** [review]

**Comment by ue71603** [comment]
VDV452 ist not working out.... trying to fix it.

**Comment by skinkie** [comment]
> VDV452 ist not working out.... trying to fix it.

Already fixed.

**Review by ue71603 — APPROVED** [review]
ok you did it.

File: xsd/NeTEx_publication.xsd [hunk 1]
```diff
@@ -7832,6 +7832,72 @@ Correct COnstraints for PointOnRoute
 			<xsd:field xpath="@id"/>
 			<xsd:field xpath="@version"/>
 		</xsd:key>
+		<!-- =====RechargingPointAssignment===(NM)===================== -->
+		<!-- =====RechargingPointAssignment unique========================== -->
+		<xsd:unique name="RechargingPointAssignment_UniqueBy_Id">
+			<xsd:annotation>
+				<xsd:documentation>Every [RechargingPointAssignment Id + Versionr] must be unique within document.</xsd:documentation>
```
Comment (skinkie) [inline xsd/NeTEx_publication.xsd:7839]:
```suggestion
				<xsd:documentation>Every [RechargingPointAssignment Id + Version] must be unique within document.</xsd:documentation>
```

File: xsd/netex_framework/netex_reusableComponents/netex_vehicleType_version.xsd [hunk 2]
```diff
@@ -1226,6 +1236,83 @@ Rail transport, Roads and Road transport
 			</xsd:element>
 			<xsd:element ref="TypeOfEquipmentRef" minOccurs="0"/>
 			<xsd:element ref="PurposeOfEquipmentProfileRef" minOccurs="0"/>
+			<xsd:element name="vehicleEquipmentProfileMembers" type="vehicleEquipmentProfileMembers_RelStructure" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Members of Vehicle Profile.  +v2.0</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_framework/netex_reusableComponents/netex_vehicleType_version.xsd:1241]:
```suggestion
					<xsd:documentation>Members of Vehicle Profile. +v2.0</xsd:documentation>
```

File: xsd/netex_part_1/part1_ifopt/netex_rechargingPointAssignment_support.xsd [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+Business and industry, Transport, Air transport , Ports and maritime transport, Public transport,
+Rail transport, Roads and Road transport
+</Category>
+					<Project>CEN TC278 WG3 SG9.</Project>
+				</Subject>
+				<Title>NeTEx RECHARGING POINT ASSIGNMENT identifier types.</Title>
+				<Type>Standard</Type>
+			</Metadata>
+		</xsd:appinfo>
+		<xsd:documentation>NeTEX: RECHARGING POINT ASSIGNMENT  identifier types.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_rechargingPointAssignment_support.xsd:57]:
```suggestion
		<xsd:documentation>NeTEX: RECHARGING POINT ASSIGNMENT identifier types.</xsd:documentation>
```

File: xsd/netex_part_1/part1_ifopt/netex_rechargingPointAssignment_support.xsd [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	<!-- ===== RECHARGING POINT ASSIGNMENT======================================================= -->
+	<xsd:simpleType name="RechargingPointAssignmentIdType">
+		<xsd:annotation>
+			<xsd:documentation>Identifier for a RECHARGING POINT ASSIGNMENT.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:restriction base="AssignmentIdType"/>
+	</xsd:simpleType>
+	<xsd:element name="RechargingPointAssignmentRef" type="RechargingPointAssignmentRefStructure" substitutionGroup="AssignmentRef">
+		<xsd:annotation>
+			<xsd:documentation>Reference to a RECHARGING POINT ASSIGNMENT</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_rechargingPointAssignment_support.xsd:68]:
```suggestion
			<xsd:documentation>Reference to a RECHARGING POINT ASSIGNMENT.</xsd:documentation>
```

File: xsd/netex_part_1/part1_ifopt/netex_rechargingPointAssignment_version.xsd [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+	</xsd:annotation>
+	<!-- =========== RECHARGING POINT ASSIGNMENT ============================================== -->
+	<xsd:group name="RechargingPointAssignmentInFrameGroup">
+		<xsd:annotation>
+			<xsd:documentation>RECHARGINGPOINT ASSIGNMENT related elements in frame.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:sequence>
+			<xsd:element name="rechargingPointAssignments" type="rechargingPointAssignments_RelStructure" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>RECHARGING POINTss in frame. +v2.0 </xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_ifopt/netex_rechargingPointAssignment_version.xsd:72]:
```suggestion
					<xsd:documentation>RECHARGING POINTss in frame. +v2.0</xsd:documentation>
```

File: xsd/netex_part_2/part2_journeyTimes/netex_coupledJourney_support.xsd [hunk 6]
```diff
@@ -245,6 +245,18 @@ Rail transport, Roads and Road transport
 	</xsd:complexType>
 	<!-- ======================================================================= -->
 	<!-- ====BLOCK======================================================== -->
+	<xsd:complexType name="blockRefs_RelStructure">
+		<xsd:annotation>
+			<xsd:documentation>A collection of one or more references to a Bkick</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_2/part2_journeyTimes/netex_coupledJourney_support.xsd:250]:
```suggestion
			<xsd:documentation>A collection of one or more references to a BLOCK.</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_support.xsd [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+	</xsd:element>
+	<xsd:complexType name="RechargingPlanRefStructure">
+		<xsd:annotation>
+			<xsd:documentation>Type for Reference to a RECHARGING PLAN.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:simpleContent>
+			<xsd:restriction base="VersionOfObjectRefStructure">
+				<xsd:attribute name="ref" type="RechargingPlanIdType" use="required">
+					<xsd:annotation>
+						<xsd:documentation>Identifier of a RechargingPlan.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_support.xsd:79]:
```suggestion
						<xsd:documentation>Identifier of a RECHARGING PLAN.</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+	</xsd:annotation>
+	<!-- ======================================================================= -->
+	<xsd:group name="RechargingPlanInFrameGroup">
+		<xsd:annotation>
+			<xsd:documentation>Elements for a RECHARGING PLAN in FRAME. +v2.0</xsd:documentation>
+		</xsd:annotation>
+		<xsd:sequence>
+			<xsd:element name="rechargingPlans" type="rechargingPlans_RelStructure" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>List of RECHARGING PLANS in frame.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd:72]:
```suggestion
					<xsd:documentation>List of RECHARGING PLANs in frame.</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="Description" type="MultilingualString" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Description of RECHARGING STEP.</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="ChargeEnergy" type="WattHoursType" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Quantity of energy for charging in Watt Hours [Wh]</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd:254]:
```suggestion
					<xsd:documentation>Quantity of energy for charging in Watt Hours [Wh].</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="ChargingDuration" type="xsd:duration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Duration of the charging process without preparation or finishing time.</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="FinalisationDuration" type="xsd:duration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Duration of the unhooking process</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd:281]:
```suggestion
					<xsd:documentation>Duration of the unhooking process.</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+		</xsd:sequence>
+	</xsd:group>
+	<xsd:group name="RechargingDurationGroup">
+		<xsd:annotation>
+			<xsd:documentation>Elements for RECHARGING STEP.</xsd:documentation>
+		</xsd:annotation>
+		<xsd:sequence>
+			<xsd:element name="PreparationDuration" type="xsd:duration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Duration of the setup process</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd:271]:
```suggestion
					<xsd:documentation>Duration of the setup process.</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+								</xsd:annotation>
+							</xsd:element>
+							<xsd:element name="Description" type="MultilingualString" minOccurs="0">
+								<xsd:annotation>
+									<xsd:documentation>Description of RECHARGING STEP.</xsd:documentation>
+								</xsd:annotation>
+							</xsd:element>
+							<xsd:element name="ChargeEnergy" type="WattHoursType" minOccurs="0">
+								<xsd:annotation>
+									<xsd:documentation>Quantity of energy for charging in Watt Hours [Wh]</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd:208]:
```suggestion
									<xsd:documentation>Quantity of energy for charging in Watt Hours [Wh].</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd [hunk 13]
```diff
@@ -0,0 +1,286 @@
...
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="RechargingProcessType" type="RechargingProcessEnumeration" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Description of RECHARGING PLAN.</xsd:documentation>
+				</xsd:annotation>
+			</xsd:element>
+			<xsd:element name="TotalChargeEnergy" type="WattHoursType" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Overall quantity of energy for charging in Watt Hours [Wh]</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd:148]:
```suggestion
					<xsd:documentation>Overall quantity of energy for charging in Watt Hours [Wh].</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+			<xsd:group ref="RechargingDurationGroup"/>
+			<xsd:element ref="JourneyRef" minOccurs="0"/>
+			<xsd:element ref="PointInJourneyPatternRef" minOccurs="0"/>
+			<xsd:element ref="VehicleTypeRef" minOccurs="0"/>
+			<xsd:element ref="VehicleRef" minOccurs="0"/>
+		</xsd:sequence>
+	</xsd:group>
+	<xsd:group name="RechargingDurationGroup">
+		<xsd:annotation>
+			<xsd:documentation>Elements for RECHARGING STEP.</xsd:documentation>
```
Comment (trurlurl) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_version.xsd:266]:
```suggestion
			<xsd:documentation>Durations of RECHARGING STEP phases.</xsd:documentation>
```

File: xsd/netex_framework/netex_reusableComponents/netex_nm_chargingEquipmentProfile_version.xsd [hunk 15]
```diff
@@ -134,21 +134,17 @@ Rail transport, Roads and Road transport
 			</xsd:element>
 			<xsd:element name="ChargingVoltage" type="VoltageType" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>CHarging voltage in Volts.
-</xsd:documentation>
+					<xsd:documentation>Charging voltage in Volts.</xsd:documentation>
```
Comment (trurlurl) [inline xsd/netex_framework/netex_reusableComponents/netex_nm_chargingEquipmentProfile_version.xsd:137]:
```suggestion
					<xsd:documentation>Charging voltage in volts.</xsd:documentation>
```

File: xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_support.xsd [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+						<xsd:documentation>Identifier of a RECHARGING STEP.</xsd:documentation>
+					</xsd:annotation>
+				</xsd:attribute>
+			</xsd:restriction>
+		</xsd:simpleContent>
+	</xsd:complexType>
+	<!-- ======================================================================= -->
+	<xsd:simpleType name="RechargingProcessEnumeration">
+		<xsd:annotation>
+			<xsd:documentation>Allowed value for RECHARGING PROCESS.</xsd:documentation>
```
Comment (trurlurl) [inline xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_support.xsd:114]:
```suggestion
			<xsd:documentation>Allowed value for RECHARGING PROCESS. +v2.0</xsd:documentation>
```

Comment (trurlurl) [reply xsd/netex_part_2/part2_vehicleService/netex_vehicleRechargingPlan_support.xsd:114]: For me it isn't completely clear where signalling "+2.0" is needed and where not...

**Review by trurlurl — CHANGES_REQUESTED** [review]

**Review by trurlurl — APPROVED** [review]

**Review by Aurige — APPROVED** [review]

**Pull request merged**